### PR TITLE
Add `S3GetObjectOperator`

### LINKED ISSUE #27811: Add  s3 object get
### Description

The s3 operator has a create object and a List keys operator, but it does not have the ability to import a single object (key).
We added the ability to import a single object.

### Use case/motivation

I want to add a function to read a single file stored in s3 on memory without saving it locally.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Lee2532**
Function that gets only one object with bucket name and bucket key
closes: [27811](https://github.com/apache/airflow/issues/27811)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by uranusjr** [comment]
Please follow the contributor guides, run linters locally with pre-commit, and fix the errors.

**Comment by Taragolis** [comment]
@Lee2532 You need to add tests. Because looks like this operator wouldn't work with default XCom backend
- boto3.resources.factory.s3.Object - not JSON serializable
- boto3.resources.factory.s3.Object - not `pickle`ble

**Comment by vincbeck** [comment]
Please also add documentation for this new operator. You likely need to update this [file](https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-amazon/operators/s3.rst)

File: docs/apache-airflow-providers-amazon/operators/s3.rst [hunk 1]
```diff
@@ -190,6 +190,21 @@ You can specify a ``prefix`` to filter the objects whose name begins with such p
     :start-after: [START howto_operator_s3_list]
     :end-before: [END howto_operator_s3_list]
 
+
+Get Amazon S3 one object
+======================
```
Comment (vincbeck) [inline docs/apache-airflow-providers-amazon/operators/s3.rst:195]:
```suggestion
Get Amazon S3 object
======================
```

File: docs/apache-airflow-providers-amazon/operators/s3.rst [hunk 2]
```diff
@@ -190,6 +190,21 @@ You can specify a ``prefix`` to filter the objects whose name begins with such p
...
+
+To one Amazon S3 object within an Amazon S3 bucket you can use
```
Comment (vincbeck) [inline docs/apache-airflow-providers-amazon/operators/s3.rst:197]:
```suggestion
To read one Amazon S3 object within an Amazon S3 bucket you can use
```

File: docs/apache-airflow-providers-amazon/operators/s3.rst [hunk 3]
```diff
@@ -190,6 +190,21 @@ You can specify a ``prefix`` to filter the objects whose name begins with such p
     :start-after: [START howto_operator_s3_list]
     :end-before: [END howto_operator_s3_list]
 
+
```
Comment (vincbeck) [inline docs/apache-airflow-providers-amazon/operators/s3.rst:193]:
```suggestion
.. _howto/operator:S3GetObjectOperator:

```

File: airflow/providers/amazon/aws/operators/s3.py [hunk 4]
```diff
@@ -757,3 +757,36 @@ def execute(self, context: Context):
         )
 
         return hook.list_prefixes(bucket_name=self.bucket, prefix=self.prefix, delimiter=self.delimiter)
+
+class S3GetObejectOperator(BaseOperator):
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/operators/s3.py:761]:
```suggestion
class S3GetObjectOperator(BaseOperator):
```

File: airflow/providers/amazon/aws/operators/s3.py [hunk 5]
```diff
@@ -757,3 +757,36 @@ def execute(self, context: Context):
...
+    """
+    Get object from `key-path` as string.
+
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/operators/s3.py:764]:
```suggestion

.. seealso::
        For more information on how to use this operator, take a look at the guide:
        :ref:`howto/operator:S3GetObjectOperator`
 
```

**Review by vincbeck — CHANGES_REQUESTED** [review]

**Comment by vincbeck** [comment]
I actually should have asked this question before commenting on this PR but what is your use case here? This operator returns the object specified in `s3_bucket` and `s3_key` but can you describe a DAG where this operator would be used? A S3 file can be big and Xcom should not be used to share large amount of data  between operators. If what you are trying to achieve is reading data from S3 bucket and send it to another source, maybe a transfer operator would be more suitable for this use case.

**Comment by potiuk** [comment]
Would be great to fix all the comments/typos.

**Comment by shubham22** [comment]
@Lee2532 - I'm wondering if you're still looking into this. Let us know if you need any help. Also, if you get chance, can you clarify the use case as @vincbeck asked?

**Comment by potiuk** [comment]
Closing then

**Pull request closed without merging**